tests/baselines/reference/privateWriteOnlyAccessorRead.js

```diff
@@ -0,0 +1,87 @@
+//// [privateWriteOnlyAccessorRead.ts]
+class Test {
+  set #value(v: { foo: { bar: number } }) {}
+  set #valueRest(v: number[]) {}
+  set #valueOne(v: number) {}
+
+  m() {
+    const foo = { bar: 1 };
+    console.log(this.#value); // error
+    this.#value = { foo }; // ok
+    this.#value = { foo }; // ok
+    this.#value.foo = foo; // error
+
+    ({ o: this.#value } = { o: { foo } }); //ok
+    ({ ...this.#value } = { foo }); //ok
+
+    ({ foo: this.#value.foo } = { foo }); //error
+    ({
+      foo: { ...this.#value.foo },
+    } = { foo }); //error
+
+    let r = { o: this.#value }; //error
+
+    [this.#valueOne, ...this.#valueRest] = [1, 2, 3];
+    let arr = [
+        this.#valueOne,
+        ...this.#valueRest
+    ];
+  }
+}
+new Test().m();
+
+
+//// [privateWriteOnlyAccessorRead.js]
+var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
+    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
+    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
+    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
+};
+var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
+    if (kind === "m") throw new TypeError("Private method is not writable");
+    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
+    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
+    return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
+};
+var __rest = (this && this.__rest) || function (s, e) {
+    var t = {};
+    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
+        t[p] = s[p];
+    if (s != null && typeof Object.getOwnPropertySymbols === "function")
+        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
+            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
+                t[p[i]] = s[p[i]];
+        }
+    return t;
+};
+var _Test_instances, _Test_value_set, _Test_valueRest_set, _Test_valueOne_set;
+class Test {
+    constructor() {
+        _Test_instances.add(this);
+    }
+    m() {
+        var _a, _b, _c;
+        const foo = { bar: 1 };
+        console.log(__classPrivateFieldGet(this, _Test_instances, "a")); // error
+        __classPrivateFieldSet(// error

```

Comment:
This is a side-effect of not cloning the receiver. It might have been lost in the conversations above but I posted [this](https://github.com/microsoft/TypeScript/pull/42458#discussion_r596977189):

> I was mainly keeping the existing code. I removed cloneNode in c461b921d93d171a49f3a38b4b6cb41c015d7fd3. It looks harmless but trailing comments don't look great. Let me know if you want to keep the change in or revert.

Reply:
Have you set the comment range on the `__classPrivateFieldSet` call using `setCommentRange`? Normally when we emit nodes, we preserve the containing node's comment start and end pos as we emit. If the container's comment start pos is the same as the current node's comment start pos, then we won't emit the comment. If the container's comment start pos is `-1`, then we'll try to emit any leading comments.

Reply:
Looks like the issue is `receiver`'s comment range. I pushed a fix but not exactly sure if this is the correct way to do it.